File: Ems Backend/src/main/java/com/cms/employeemanagement/service/HolidayService.java
package com.cms.employeemanagement.service;

import java.util.List;

import com.cms.employeemanagement.entity.Holiday;

public interface HolidayService {

    Holiday addHoliday(Holiday holiday);

    Holiday updateHoliday(Long id, Holiday holiday);

    void deleteHoliday(Long id);

    List<Holiday> getAllHolidays();

    List<Holiday> getUpcomingHolidays();

    Holiday getHolidayById(Long id);

}

File: Ems -Frontend/src/services/holidayService.js
import axios from "axios";

const API_URL = "http://localhost:9090/api/holidays";

// Get All Holidays
export const getAllHolidays = () => {
    return axios.get(API_URL);
};

// Get Upcoming Holidays
export const getUpcomingHolidays = () => {
    return axios.get(`${API_URL}/upcoming`);
};

// Get Holiday By Id
export const getHolidayById = (id) => {
    return axios.get(`${API_URL}/${id}`);
};

// Add Holiday
export const addHoliday = (holiday) => {
    return axios.post(API_URL, holiday);
};

// Update Holiday
export const updateHoliday = (id, holiday) => {
    return axios.put(`${API_URL}/${id}`, holiday);
};

// Delete Holiday
export const deleteHoliday = (id) => {
    return axios.delete(`${API_URL}/${id}`);
};